test/miniyaml.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+)
+
+var keyValue *regexp.Regexp = regexp.MustCompile(`^(\s*)(\S+):\s*(\S.+?)\s*$`)
+var keyOnly *regexp.Regexp = regexp.MustCompile(`^(\s*)(\S+):\s*$`)
+var listItem *regexp.Regexp = regexp.MustCompile(`^(\s*)-\s*(.+?)\s*$`)
+
+type YamlDoc map[string]interface{}
+type YamlList []string
+
+type stackItem struct {
+	item   interface{}
+	key    string
+	indent int
+}
+
+// Only parses key:val pairs (both strings) and lists of strings
+func ParseYaml(rdr io.Reader) (YamlDoc, error) {
```

Comment:
ach, sorry, i wish i'd caught it before you put all the time in on that - sorry! thanks for making the change.